### PR TITLE
#153 Added minimesos uninstall command

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: minimesos/src/main/java/com/containersol/minimesos/marathon/MarathonContainer.java [hunk 1]
```diff
@@ -78,6 +78,19 @@ public void setZooKeeper(ZooKeeper zooKeeper) {
...
 
     @Override
+    public void deleteApp(String app) {
+        try {
+            HttpResponse<JsonNode> deleteResponse = Unirest.delete(getServiceUrl().toString() + END_POINT_EXT + "/" + app).header(HEADER_ACCEPT, APPLICATION_JSON).asJson();
+            deleteResponse.getBody().getObject();
+            if (!(deleteResponse.getStatus() == HttpStatus.SC_OK)) {
+                throw new MinimesosException("Could not delete app: " + app);
+            }
+        } catch (UnirestException e) {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/marathon/MarathonContainer.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        Marathon marathon = cluster.getMarathon();
+        if (marathon == null) {
+            throw new MinimesosException("Marathon container is not found in cluster " + cluster.getClusterId());
+        }
+
+        try {
+            marathon.deleteApp(app);
+        } catch (MinimesosException e) {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "uninstall";
+    }
+
+    @Override
+    public void execute() throws MinimesosException {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java]: MINOR Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by containersol** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 2 critical
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return "uninstall";
+    }
+
+    @Override
+    public void execute() {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUninstall.java]: MINOR Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by containersol** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 2 critical
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by sadovnikov** [comment]
Approved

**Pull request merged**